doc/filters/merge.rst

```diff
@@ -42,6 +42,25 @@ overridden.
         
 .. note::
 
-    Internally, Twig uses the PHP `array_merge`_ function.
+    Internally, Twig uses the PHP `array_merge`_ function, which means that it will loose any numerical index after a merge, unless the numerical index is concatenated with something to make a string.
```

Comment:
which means it is **not** a numerical index anymore at all, but a string index :smile: